### DISCUSSION TITLE
Tailwindcss/nesting and postcss-nesting

### DISCUSSION THREAD

**Discussion by joggienl** [vercel/next.js#33237] [Help]
Does anyone have an idea on how to load postcss-nesting via tailwindcss/nesting, as described via the tailwindcss documentation?

The [tailwindcss documentation](https://tailwindcss.com/docs/using-with-preprocessors#nesting) tells you to include this inside the `postcss.config.js`:

```js
module.exports = {
  plugins: [
    require('postcss-import'),
    require('tailwindcss/nesting')(require('postcss-nesting')),
    require('tailwindcss'),
    require('autoprefixer'),
  ]
}
```
But NextJs [does not allow](https://nextjs.org/docs/advanced-features/customizing-postcss-config#customizing-plugins) to setup postcss like that.

> Do not use require() to import the PostCSS Plugins. Plugins must be provided as strings.

So, to configure it (what works):

```js
// postcss.config.js
module.exports = {
  plugins: [
    'tailwindcss/nesting',
    'tailwindcss',
    'postcss-flexbugs-fixes',
    [
      'postcss-preset-env',
      {
        autoprefixer: {
          flexbox: 'no-2009',
        },
        stage: 3,
        features: {
          'custom-properties': false,
          'nesting-rules': false,
        },
      },
    ],
  ],
};

```

But at this point I have no idea on how to load the `postcss-nesting` like `require('tailwindcss/nesting')(require('postcss-nesting'))` (I do not know if there is an config we could pass to make it work).

I could just add `postcss-nesting` in stead of `tailwindcss/nesting`, but I am not sure if this will eventually break tailwindcss stuff because the nesting isn's actually loaded properly..

Thanks in advance!

**Comment by faiz-ansari-afk** [comment]
I'm also facing this issue,
but somehow [Filip Jerga's repo](https://github.com/Jerga99/next-ecommerce-course) had implemented this.
You can find out in this file [Grid Component](https://github.com/Jerga99/next-ecommerce-course/blob/main/components/ui/Grid/Grid.module.css)
